### PR TITLE
p2p: Rename fBlocksOnly, Add test

### PR DESCRIPTION
**Pull request by maflcko**
`fBlocksOnly` has several issues:
* The name is confusing
* It is untested

Fix both.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Fun fact: This is covered by the fuzz tests, but obviously they don't check the logic level of the code. https://marcofalke.github.io/btc_cov/fuzz.coverage/src/net_processing.cpp.gcov.html

Functional coverage: https://marcofalke.github.io/btc_cov/total.coverage/src/net_processing.cpp.gcov.html

**Comment by MarcoFalke** [comment]
There are some other issues with this code, which will be fixed in follow-ups, because they change the P2P protocol.

File: src/net_processing.cpp [hunk 1]
```diff
@@ -2909,13 +2909,13 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
             return;
         }
 
-        // We won't accept tx inv's if we're in blocks-only mode, or this is a
+        // Reject tx INVs when the -blocksonly setting is enabled, or this is a
         // block-relay-only peer
-        bool fBlocksOnly = m_ignore_incoming_txs || (pfrom.m_tx_relay == nullptr);
+        bool reject_tx_invs{m_ignore_incoming_txs || (pfrom.m_tx_relay == nullptr)};
```
Comment (naumenkogs) [inline src/net_processing.cpp:2914]: Why not calling it ignore? Reject sounds like something proactive.

Comment (MarcoFalke) [reply src/net_processing.cpp:2914]: It will disconnect the peer, which seems proactive and stronger than "ignore"

Comment (naumenkogs) [reply src/net_processing.cpp:2914]: Good point, I forgot it disconnects. Still unsure "reject" captures the meaning tho.
I don't have a better alternative to suggest at this point so feel free to ignore.

Comment (MarcoFalke) [reply src/net_processing.cpp:2914]: Happy to change when there is a better name. Resolving for now.

**Review by vincenzopalazzo — APPROVED** [review]
ack https://github.com/bitcoin/bitcoin/pull/22955/commits/fac66d0a39cb0b4bc565b57087ba84dd932e9b6d

**Comment by laanwj** [comment]
Code review ACK fa66a7d732f9fd60d72f22087f0d5aadf3064bfb

**Pull request merged**